ibis/expr/operations/relations.py

```diff
@@ -1,671 +1,466 @@
-                if not (dtype := predicate.dtype).is_boolean():
-                    raise com.IbisTypeError(f"Invalid predicate dtype: {dtype}")
-            elif not shares_some_roots(predicate, table):
-                raise com.RelationError("Predicate doesn't share any roots with table")
-
-        super().__init__(
-            table=table,
-            selections=tuple(map(_add_alias, selections)),
-            predicates=predicates,
-            sort_keys=sort_keys,
-            **kwargs,
-        )
-
-    @annotated
-    def order_by(self, keys: VarTuple[SortKey]):
-        from ibis.expr.analysis import shares_all_roots, sub_immediate_parents
-
-        if not self.selections:
-            if shares_all_roots(keys, table := self.table):
-                sort_keys = tuple(self.sort_keys) + tuple(
-                    sub_immediate_parents(key, table) for key in keys
-                )
-
-                return Selection(
-                    table,
-                    self.selections,
-                    predicates=self.predicates,
-                    sort_keys=sort_keys,
-                )
+class Namespace(Concrete):
+    database: Optional[str] = None
+    schema: Optional[str] = None
 
-        return Selection(self, [], sort_keys=keys)
 
-    @attribute
-    def _projection(self):
-        return Projection(self.table, self.selections)
+@public
+class DatabaseTable(PhysicalTable):
+    schema: Schema
+    source: Any
+    namespace: Namespace = Namespace()
 
 
 @public
-class DummyTable(Relation):
-    # TODO(kszucs): verify that it has at least one element: Length(at_least=1)
-    values: VarTuple[Value[dt.Any]]
-
-    @attribute
-    def schema(self):
-        return Schema({op.name: op.dtype for op in self.values})
+class InMemoryTable(PhysicalTable):
+    schema: Schema
+    data: TableProxy
 
 
 @public
-class Aggregation(Relation):
-    table: Relation
-    metrics: VarTuple[Scalar] = ()
-    by: VarTuple[Column] = ()
-    having: VarTuple[Scalar[dt.Boolean]] = ()
-    predicates: VarTuple[Value[dt.Boolean]] = ()
-    sort_keys: VarTuple[SortKey] = ()
-
-    def __init__(self, table, metrics, by, having, predicates, sort_keys):
-        from ibis.expr.analysis import shares_all_roots, shares_some_roots
-
-        # All non-scalar refs originate from the input table
-        if not shares_all_roots(metrics + by + having + sort_keys, table):
-            raise com.RelationError(
-                "Selection expressions don't fully originate from "
-                "dependencies of the table expression."
-            )
+class SQLQueryResult(Relation):
+    """A table sourced from the result set of a select query."""
 
-        # invariant due to Aggregation and AggregateSelection requiring a valid
-        # Selection
-        assert all(shares_some_roots(predicate, table) for predicate in predicates)
+    query: str
+    schema: Schema
+    source: Any
+    fields = FrozenDict()
```

Comment:
This must be `values` instead of `fields`, doesn't have a test case for it just in the SQL backends, perhaps we could move the entire op there?

Reply:
Updated.